### ISSUE TITLE
Improve behavior around bad credentials

### ISSUE THREAD

**Issue by mrocklin** [coiled/coiled-issues#45]
If you've logged in again then old config files that hold a token are made invalid.  In this case users get an non-informative error.

Either we should improve this error, or we should consider going through the signin process again, but now forgetting the provided token.

cc @marcosmoyano @jrbourbeau

**Comment by TomAugspurger** [comment]
Piggy-backing on this, I think I was seeing issues with my config.

```yaml
# file: ~/.config/dask/coiled.yaml
coiled:
  # server: https://beta.coiledhq.com
  server: https://beta.coiled.io
  token: <token>
  user: tomaugspurger
```

I had the (old?) server URL beta.coiledhq.com and got some message about "Authentication not provided" when doing things like `coiled login --token <token>`.

**Comment by dantheman39** [comment]
Fixed in release 0.2.23
